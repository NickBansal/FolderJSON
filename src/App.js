import React, { Component } from 'react';
import './App.css';
import Folders from './Components/Folders'

class App extends Component {

  state = {
    folderItemElement: 0,
    toggleInformation: false
  }

  render() {
    const { folderItemElement, toggleInformation } = this.state
    return (
      <div className="App">
        <Folders 
        folderItemElement={folderItemElement}
        toggleInformation={toggleInformation}
        handleClick={this.handleClick}/>
      </div>
    );
  }

  handleClick = index => {
    this.setState({
      folderItemElement: index,
      toggleInformation: !this.state.toggleInformation
    })
  }

}

export default App;
