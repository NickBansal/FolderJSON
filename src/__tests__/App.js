import '../setupTest'
import React from 'react'
import App from '../App'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('<App />', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallow(<App />)
    })
    it('Component matches the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('Changes the folderItemElement and toggleInformation in state on click handler', () => {
        wrapper.instance().handleClick(0)
        expect(wrapper.state().folderItemElement).toBe(0)
        expect(wrapper.state().toggleInformation).toBe(true)
        wrapper.instance().handleClick(1)
        expect(wrapper.state().folderItemElement).toBe(1)
        expect(wrapper.state().toggleInformation).toBe(true)
        wrapper.instance().handleClick(1)
        expect(wrapper.state().toggleInformation).toBe(false)
        wrapper.instance().handleClick(2)
        expect(wrapper.state().folderItemElement).toBe(2)
    })
    it('Changes the state with the handleSubmit instance', () => {
        wrapper.instance().handleSubmit('filter')
        expect(wrapper.state().filter).toBe('filter')
        wrapper.instance().handleSubmit('random')
        expect(wrapper.state().filter).toBe('random')
    })
    it('Changes the state with the sortFolder instance', () => {
        wrapper.instance().sortFolders('name')
        expect(wrapper.state().sortValue).toBe('name')
        expect(wrapper.state().toggleInformation).toBe(false)
        expect(wrapper.state().columnReverse).toBe(false)
        wrapper.instance().sortFolders('name')
        expect(wrapper.state().columnReverse).toBe(true)
    })
})