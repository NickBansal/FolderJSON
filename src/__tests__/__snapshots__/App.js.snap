// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Component matches the snapshot 1`] = `
<div
  className="App"
>
  <SortButtons
    sortFolders={[Function]}
  />
  <FilterForm
    handleSubmit={[Function]}
  />
  <FoldersOrFiles
    columnReverse={false}
    filter=""
    handleClick={[Function]}
    sortValue=""
    toggleInformation={false}
  />
</div>
`;
